GuideBuiltforUbuntuDebianGraftDockercontainer.md

```diff
@@ -0,0 +1,325 @@
+when in wallet type: seed and follow prompt (no password just press enter)
+
+Once graftnoded fully synced and stake in wallet, kill graft_server process to speed up process.  
+ Kill graft_server process:
+````bash
+ps ax | graft_server
+````
+grab process ID = Furthest to the left of the line showing below:
+````bash
+4002 ?        Sl     1:32 graft_server --log-file supernode.log --log-level 1 > out.log 2>&1
+````
+In this example use = 4002
+````bash
+kill -9 4002
+````
+graft_server will start automatically again.
+
+Viewing logs: (-n {150} - 150 equals lines to view – adjust accordingly)
+
+Graft_server logs:
+````bash
+cd /home/graft-sn/supernode
+tail -f -n 150 supernode.log
+````
+Graftnoded logs from anywhere:
+````bash
+tail -f -n 150 /$HOME/.graft/testnet/graft.log
+````
+Checking locally inside container if supernode list is being generated:
+Install curl:
+````bash
+apt update
+apt install curl -y
+curl 127.0.0.1:28690/debug/supernode_list/0
+````
+
+Do not forget to open ports for 28690 and 28680 on VPS or port forward if hosting locally.
+
+Test list from external:
+````bash
+<external-IP>:28690/debug/supernode_list/0
+```` 
+## Docker commands
+
+Stop and Start container (use name from docker run command)
+````bash
+sudo docker stop graft
+sudo docker start graft
+````
+
+To setup docker so that container starts automatically if machine restarted:
+
+On running container after exiting container - just type exit when logged in to exit
+````bash
+sudo docker container update --restart unless-stopped graft
+````
+graft in above is the --name used in the docker run command
+
+
+## Optional:
+(Beware of using below on mainnet is much safer to download Blockchain youself | also please consult @jagerman42 before using 
+any watch-only-wallets downloads on mainnet):
+
+Before running the image download and docker run step, download provided blockchain directory.
+````bash
+sudo apt update
+sudo apt install unzip -y
+mkdir $HOME/.graft		# (if not done already)
+wget "https://www.dropbox.com/s/b55s59bluvp8s1z/graft_bc_testnet_bkp_17Nov18.zip" -P /tmp
+unzip /tmp/graft_bc_testnet_bkp_17Nov18.zip '.graft/testnet/*' -d $HOME/
+````
+
+Go back to: [Download image and run container after with mounted volume ](https://github.com/mv1879/docs/blob/master/Dockers%20by%20Fez.md#add-docker-repo)
+
+## ADVANCED
+
+Recommended Docker container updates
+Restart container automatically on reboot of server:
+````bash
+sudo docker container update --restart unless-stopped graft
+````
+Limit cpu usage to 2 cores: (adjustable)
+````bash
+sudo docker container update --cpus=2 graft
+````
+
+Download Watch only wallets:
+Login to already running container:
+````bash
+sudo docker exec -ti graft /bin/bash
+mkdir -p ~/.graft/supernode/data/{watch-only-wallets,stake-wallet} && cd ~/.graft/supernode/data && curl -s https://rta.graft.observer/lmdb/watch-only-wallets.tar | tar xvf -
+````
+Kill graft_server process:
+````bash
```

Comment:
Like above: `killall graft_server` is simpler

Reply:
Noted will adjust